backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/switch/BpkSwitch.kt

```diff
@@ -73,15 +75,20 @@ fun BpkSwitch(
     Row(
         verticalAlignment = Alignment.CenterVertically,
         horizontalArrangement = Arrangement.SpaceBetween,
-        modifier = modifier.applyIf(onCheckedChange != null) {
-            toggleable(
-                value = checked,
-                role = Role.Switch,
-                interactionSource = interactionSource,
-                indication = null,
-                onValueChange = onCheckedChange!!,
-            )
-        },
+        modifier = modifier
+            .applyIf(onCheckedChange != null) {
+                toggleable(
```

Comment:
i think actually what is missing i that we're not passing `enabled` through to the `toggleable` modifier

Reply:
how do you mean? i've tried passing it through as a parameter but it doesn't seem to have any effect, also requires the OnCheckedChange to be able to affect the disabled switches as it's not a required parameter :/

Reply:
is this the intended behaviour you're meaning? my solution isn't the best but i can't think of.a way of using the selected semantics to improve it 🤔 <video src="https://github.com/Skyscanner/backpack-android/assets/19375170/467cc629-98f3-4e1d-bc11-7f8f4401667b"/>